src/index.ts

```diff
@@ -1,33 +1,26 @@
 import path from 'path'
+import {StringDecoder} from 'string_decoder'
 import {createCompiler} from '@mdx-js/mdx'
-import {babel as rollupBabel} from '@rollup/plugin-babel'
-import commonjs from '@rollup/plugin-commonjs'
-import json from '@rollup/plugin-json'
-import {nodeResolve} from '@rollup/plugin-node-resolve'
-import babelPresetReact from '@babel/preset-react'
-import babelPresetEnv from '@babel/preset-env'
-import babelPresetTypeScript from '@babel/preset-typescript'
 import matter from 'gray-matter'
-import type {OutputOptions, RollupOptions} from 'rollup'
-import {rollup} from 'rollup'
-import {terser} from 'rollup-plugin-terser'
+import {build as bundle, Plugin, BuildOptions} from 'esbuild'
+import nodeResolve from '@esbuild-plugins/node-resolve'
+import {globalExternals} from '@fal-works/esbuild-plugin-global-externals'
+import type {ModuleInfo} from '@fal-works/esbuild-plugin-global-externals'
+
+type ESBuildOptions = BuildOptions & {write: false}
 
 async function bundleMDX(
   mdxSource: string,
   {
     files = {},
     remarkPlugins = [],
-    rollup: {
-      getInputOptions = (options: RollupOptions) => options,
-      getOutputOptions = (options: OutputOptions) => options,
-    } = {},
+    esbuildOptions = (options: ESBuildOptions) => options,
+    globals = {},
   }: {
     files?: Record<string, string>
     remarkPlugins?: Array<unknown>
-    rollup?: {
-      getInputOptions?: (options: RollupOptions) => RollupOptions
-      getOutputOptions?: (options: OutputOptions) => OutputOptions
-    }
+    esbuildOptions?: (options: ESBuildOptions) => ESBuildOptions
```

Comment:
I wonder if we need to give the users access to the plugins here?

I know plugin order matters so its concievable that they may want to add a plugin between some of the ones we use.

Something like:

```typescript
esbuildOptions: (options, {globalExternals, nodeResolve, inMemoryPlugin}) => {
  options.plugins = [
    globalExternals(/* globals */),
    nodeResolve(/* copy config from mdx-bundler */),
    userPlugin,
    inMemoryPlugin
  ]
}
```

I maybe thinking too much into this...

Reply:
I think they could just copy/paste what we're doing if they need to do that.

Reply:
ok, I'll leave it as it is.